design/rest-api-guideline.md

```diff
@@ -454,7 +454,29 @@ Keycloak does not provide rate limiting capabilities and for such it is necessar
 
 ### Filtering
 
-TODO
+Sometimes, we need more control over the content that will return from the server, so that only a subset of the data is available in the response. For example, a list user's groups or permissions. 
+
+The Keycloak admin REST API allows to search for users based on `e-mail`, `first name`, `lastname` and other optional parameters. However, such implementation is limited to a particular context (e.g, collection of users).
+
+Filtering can be implemented as a query parameter, like we already did for the current REST API, with some enhancements. To filter on a field, should be as simple as the following request:
```

Comment:
I think that *can* in "Filtering can be implemented" is not properly recommending the usage of query parameters for filtering. Shouldn't be `should` instead. I think we are using should in most places to indicate that "there may exist valid reasons in particular circumstances to ignore a particular item, but the full implications must be understood and
   carefully weighed before choosing a different course".